backend/apid/actions/users.go

```diff
@@ -74,17 +78,32 @@ func (a UserController) CreateOrReplace(ctx context.Context, user *corev2.User)
 		return NewError(InvalidArgument, err)
 	}
 
-	// Validate password
-	if err := user.ValidatePassword(); err != nil {
-		return NewError(InvalidArgument, err)
-	}
+	// Determine if a hashed and/or cleartext password was provided
+	if user.Password != "" && user.PasswordHash != "" {
```

Comment:
I think there should be a check to see if these things are equal.

Reply:
We actually do it just below with `bcrypt.CheckPassword(user.PasswordHash, user.Password)`!

Reply:
So correct me if I'm wrong, but `bcrypt.CheckPassword` will compare a hash with a plaintext password. But I think we should check to see if `Password` contains a hash that is equal to `PasswordHash`.

Reply:
Right, but `Password` contains here a cleartext password; my understanding is that by design, you can't compare two hashes one of these need to be sent as cleartext.

Reply:
Oh, of course, since this is CreateOrUpdate, it's not pulling anything from the db.